framework/src/st_things/things_stack/framework/things_data_manager.c

```diff
@@ -1971,12 +1969,9 @@ bool dm_is_rsc_published(void)
 static things_resource_s *register_resource(things_server_builder_s *p_builder, struct things_resource_info_s *resource, const char *id)
 {
 	things_resource_s *ret = NULL;
-	char res_uri[MAX_URI_LENGTH] = { 0 };
-
 	if (NULL != resource) {
 		THINGS_LOG_D(TAG, "RESOURCE TO REGISTER : %s", resource->uri);
-
-		memset(res_uri, 0, (size_t) MAX_URI_LENGTH);
+		char res_uri[MAX_URI_LENGTH] = { 0 };
 		strncat(res_uri, resource->uri, MAX_URI_LENGTH);
```

Comment:
MAX_URI_LENGTH - 1

Reply:
Better to set the size of URI to `MAX_URI_LENGTH + 1`
(This can accommodate a URI whose length is exactly 64).

Note: It's existing code and not reported by static analysis fix.